grakn-graph/src/main/java/ai/grakn/factory/SystemKeyspace.java

```diff
@@ -82,10 +83,11 @@
     protected static final Logger LOG = LoggerFactory.getLogger(SystemKeyspace.class);
 
     private static final ConcurrentHashMap<String, Boolean> openSpaces = new ConcurrentHashMap<>();
-    private static final AtomicBoolean isFactorySet = new AtomicBoolean(false);
-    private static InternalFactory factory;
+    static final AtomicBoolean factoryBeingInstantiated = new AtomicBoolean(false);
```

Comment:
Unfortunately this is the best way that I could come up with and I hoped that package-private might not be so nasty. 

@aelred Do you have any ideas how we can better test this?